tests/mocktests.js

```diff
@@ -0,0 +1,48 @@
+var test = require('tape')
+var fs = require('fs')
+var os = require('os')
+var mockdata = require('./mockdata')
+
+var orig = {
+  platform: os.platform,
+  stat: fs.stat,
+  readFile: fs.readFile
+}
+
+var currentData
+
+os.platform = function() {
+  return currentData.platform
+}
+
+fs.stat = function(file, callback) {
+  process.nextTick(function() {
+    if (!currentData.file[file])
+      return callback(new Error())
+    callback(null, { isFile: function() { return true }})
+  })
+}
+
+fs.readFile = function(file, enc, callback) {
+  process.nextTick(function() {
```

Comment:
always good practice to make functions that take callbacks actually be async so the user doesn't need to care about the order of call:

```js
fs.readFile('foo', function (err, data) {
  console.log('I read the file')
})
console.log('I am about to read the file')
```

```
I am about to read the file
I read the file
```
vs

```
I read the file
I am about to read the file
```